File: makeCommaTest.php
<?php
class MakeCommaTest
{
  private function _makeComma($num)
  {
    $isChk = false;
    if (strpos($num, "-") !== false) //strpos — 문자열이 처음 나타나는 위치 , -기호가있으면
    {
      $isChk = true;
    }

    if (strpos($num, ".") > 0) //소수점 처리
    {
      $pos = strpos($num,"."); //소수점위치
      $strTail = substr($num, $pos); //소수점위치부터 담기
      $num = substr($num, 0, $pos); //소수점 위치 직전까지 담기
    }

    for($i=0; $i<strlen($num); $i++)
    {
      if ($i!=0 && $i%3==(strlen($num)%3)) //strlen($num)%3 => 콤마위치, 콤마찍을 위치이면,
      {
         if (!$isChk || $i!=1) //-,200,000
          $str .= ",";
      }
      $str .= substr($num, $i, 1);
    }

   // echo $str.$strTail;
    return $str.$strTail;
  }

  public function index()
  {
    echo $this->_makeComma(-323425.24); //-323,425.24
    echo $this->_makeComma(2000000); //2,000,000
    echo $this->_makeComma(2000); //-200,000
  }
	
}

File: MakeCommaTest.java
package test12;

public class MakeCommaTest {
	
	//숫자 형태의 문자열을 콤마가 포함된 금액 표기식 문자열로
	public static void main(String args[]){
		
		long sTime = System.currentTimeMillis();		
		System.out.println(makeComma("-234345523432223252323.5562345")); //-234,345,523,432,223,252,323.5562345
		long eTime = System.currentTimeMillis() - sTime;
		System.out.println(eTime + " ms");
		
		System.out.println(makeComma("20000000")); //20,000,000
		System.out.println(makeComma("2000000")); //2,000,000		
	}
	
	private static String makeComma(String num) {
		
		int cnt = 0;
		//문자열 담기 위한 변수
		String str = "";
		//소수점 이하 담기 위한 변수
		String strTail = "";
		//비교 기호
		String op = "-, +";
		
		//소수점		
		if (num.indexOf(".") >= 0) {
			strTail = num.substring(num.indexOf(".")); //소수점이 있으면(소수점위치반환)소수점 부터 끝까지 추출하여 strTail 변수에 저장
			num = num.substring(0, num.indexOf("."));//소수점이 있으면 숫자를 0부터 소수점 직전까지 가지고 와서 num 변수에 저장		
		}
		
		boolean isOp = false;
		if (op.indexOf(num.charAt(0)) >= 0) // +,- 가 있으면
			isOp = true;
		
		for (int i=num.length()-1; i>=0; i--) //charat i=0 시작
		{
			cnt++;
			System.out.println(str);
			if ( cnt%3 == 0 && i != 0 ) { //,200,000 처리			
				if (isOp && i == 1) str = num.charAt(i)+str; //-,325,436,454,323,425.24 처리
				else str = ","+num.charAt(i)+str;
			} else {
				str = num.charAt(i)+str;
			}
		}
		
		//System.out.println(str + strTail);
		return str + strTail;
	}	

}

